lib/broccoli/ember-app.js

```diff
@@ -218,28 +218,38 @@ EmberApp.prototype._initOptions = function(options, isProduction) {
   this.options.babel.sourceMaps = false;
 
   this.options.trees = defaults(this.options.trees, {
-    app:       new WatchedDir('app'),
-    tests:     new WatchedDir('tests'),
+    app:       new WatchedDir(this._resolveLocal('app')),
+    tests:     new WatchedDir(this._resolveLocal('tests')),
 
     // these are contained within app/ no need to watch again
     // (we should probably have the builder or the watcher dedup though)
-    styles:    new UnwatchedDir('app/styles'),
-    templates: existsSync('app/templates') ? new UnwatchedDir('app/templates') : null,
+    styles:    new UnwatchedDir(this._resolveLocal('app/styles')),
+    templates: existsSync(this._resolveLocal('app/templates')) ? new UnwatchedDir(this._resolveLocal('app/templates')) : null,
 
     // do not watch vendor/ or bower's default directory by default
     bower: this.project._watchmanInfo.enabled ? this.bowerDirectory : new UnwatchedDir(this.bowerDirectory),
-    vendor: existsSync('vendor') ? new UnwatchedDir('vendor') : null,
+    vendor: existsSync(this._resolveLocal('vendor')) ? new UnwatchedDir(this._resolveLocal('vendor')) : null,
 
-    public: existsSync('public') ? new WatchedDir('public') : null
+    public: existsSync(this._resolveLocal('public')) ? new WatchedDir(this._resolveLocal('public')) : null
   });
 
   this.options.jshintrc = defaults(this.options.jshintrc, {
     app: this.project.root,
-    tests: path.join(this.project.root, 'tests'),
+    tests: this._resolveLocal('tests'),
   });
 };
 
 /**
+  Resolves a path relative to the project's root
+
+  @private
+  @method _resolveLocal
+*/
+EmberApp.prototype._resolveLocal = function(to) {
```

Comment:
the current implementation has a branch, `join(x,to) || to`  although i', unclear when falsy case will be taken. I suspect if there is such a scenario, it should also be added to the test.